pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -99,10 +101,25 @@ func (c *EtcdCertSignerController) syncAllMasters(recorder events.Recorder) erro
 
 	errs := []error{}
 	for _, node := range nodes {
-		if err := c.createSecretForNode(node, recorder); err != nil {
+		nodeInternalIPs, err := dnshelpers.GetInternalIPAddressesForNodeName(node)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if err := c.checkSecrets(node, nodeInternalIPs, recorder); err != nil {
+			// if any of the secrets are not present, create secrets for the node
+			if err := c.createSecretForNode(node, nodeInternalIPs, recorder); err != nil {
+				errs = append(errs, err)
+			}
+			continue
```

Comment:
nit: Is it your intent that this `continue` exits regardless of whether the secrets are successfully created?

Reply:
yes. If the secrets are not created the `len(errs)` would be greater than zero, and hence exits the function.

Reply:
Aha, re-reading my question it wasn't clear what I was asking. Maybe add a comment to the `continue` indicating something like 'No need to validate IP addresses - either the secrets remain in an invalid state or they have been created with the correct ip addresses'?

Reply:
We want to validate the ip address, only if we already found extant certificates.

If we just created the certificates, it is assumed they are created with the right ip address. I will add a comment.